app/services/DateService.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import java.time.LocalDate
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class DateService @Inject() extends DateServiceInterface {
+
+  def getCurrentDate: LocalDate = LocalDate.now()
+
+}
+
+trait DateServiceInterface {
```

Comment:
this might create a confusion in the future; trait's some properties of Java interfaces but really different in the actual usage; if you can remove this suffix pls